pandas/tests/frame/methods/test_astype.py

```diff
@@ -584,6 +583,20 @@ def test_astype_ignores_errors_for_extension_dtypes(self, df, errors):
             with pytest.raises((ValueError, TypeError), match=msg):
                 df.astype(float, errors=errors)
 
+    @pytest.mark.parametrize("errors", ["raise", "ignore"])
```

Comment:
*may* want to consider using a shared function as the impl to avoid repeating things. though maybe not worth it (e.g. just copy in the cases where this is needed if there are not that many)

Reply:
do we have a fixture for errors? if not can u create one

Reply:
Alternatively, could you parametrize the test above, but have the dtype be a separate param, so it can be skipped for arrow_string?

Something like

```
    @pytest.mark.parametrize(
        "data, dtype",
        [
            (["x", "y", "z"], "string"),
            pytest.param((["x", "y", "z"], "string_arrow"), mark=..skip...),
            (["x", "y", "z"], "category"),
            (3 * [Timestamp("2020-01-01", tz="UTC")], None),
            (3 * [Interval(0, 1)], None),
        ],
    )
```

and then inside the test create the dataframe (`df = DataFrame(Series(data, dtype=dtype))`). 

That could also avoid having to duplicate the full test body.

Reply:
(although for the other tests where pd.array is used in the parametrization, this might get a bit more annoying to do it in this way ..)

Reply:
> do we have a fixture for errors? if not can u create one

we have no fixture yet in conftest.py (common arguments section) for errors

we have 4 tests parameterized on errors...

`test_astype_ignores_errors_for_extension_dtypes` in `pandas/tests/frame/methods/test_astype.py` and `pandas/tests/series/methods/test_astype.py` parameterised on `["raise", "ignore"]`

`test_custom_dateparsing_error` in `pandas/tests/io/test_sql.py` parameterised on `["ignore", "raise", "coerce"]`

and 

`test_coerce_outside_ns_bounds` in `pandas/tests/tslibs/test_array_to_datetime.py` parameterised on ["coerce", "raise"]

will be pushing a commit shortly with @jorisvandenbossche suggested changes to avoid duplication so doing anything with errors parameterisation is now outside the scope of this PR.

I don't think worth opening a separate PR for that yet, especially considering the differences in the parameterisation

Reply:
> (although for the other tests where pd.array is used in the parametrization, this might get a bit more annoying to do it in this way ..)

the other two tests have simpler code so duplication is probably better than complex setup

test_astype_ignores_errors_for_extension_dtypes is already duplicated in frame and series methods modules so makes more sense to avoid more duplication there.

Reply:
> > (although for the other tests where pd.array is used in the parametrization, this might get a bit more annoying to do it in this way ..)
> 
> the other two tests have simpler code so duplication is probably better than complex setup
> 
> test_astype_ignores_errors_for_extension_dtypes is already duplicated in frame and series methods modules so makes more sense to avoid adding more duplication there. have updated with suggestion.